backend/src/main/java/com/rota/api/StaffController.java

```diff
@@ -168,4 +169,18 @@ public Staff updateStaff(
     staffService.removeStaff(id);
     return staffService.getActiveStaff();
   }
+
+  /**
+   * Endpoint for the manager to add staff member's engagement.
+   */
+  @PostMapping("/staff/addEngagement")
+  @ApiOperation(value = "Allows manager to add an engagement for a certain user", authorizations = {
+      @Authorization(value = "Bearer") })
```

Comment:
How are we distinguishing manager / regular authenticated user?

Reply:
All `/staff` endpoints require manager authentication.

Reply:
I wonder if we should rethink the naming convention in the back-end too, and change `Staff` to `Employee` and clean up our endpoint naming or at least get it more consistent.

For example not all of our endpoints use verbs and some do, so for example with engagements we could use this pattern:

`GET /staff` returns all staff
`GET /staff/` returns staff with ID 1
`POST /staff` adds new staff object
`DELETE /staff/` removes staff object

`GET /staff/1/engagements` gets all staff 1's engagements
`POST /staff/1/engagements` adds an engagement to staff 1

and so on. What are peoples thoughts?

Reply:
We agreed to change /staff to /employee everywhere after the front-end cleanup PRs are merged.
Your proposed structure makes sense, the only wrinkle is that DELETE doesn't actually delete staff member from the database, so PUT might make more sense.

Reply:
So when we set inactive we will treat it more as update to inactive I guess?

Reply:
Addressed in #160

Reply:
I agree with @emab I think.

Maybe need to review the api a little. If the employee entity cannot be retrieved after DELETE/PUT then it should be DELETE request.

Reason being that clients are not aware of underlying data or how api handles deleting and does not need to know. They don't care about inactive property.